stellar/deploy-contract.js

```diff
@@ -8,12 +8,77 @@ const { getDomainSeparator, getChainConfig } = require('../common');
 const { prompt, validateParameters } = require('../common/utils');
 const { weightedSignersToScVal } = require('./type-utils');
 const { ethers } = require('hardhat');
-const { readFileSync } = require('fs');
+const { writeFileSync, readFileSync } = require('fs');
+const fetch = require('node-fetch');
+const path = require('path');
 const {
     utils: { arrayify, id },
 } = ethers;
 require('./cli-utils');
 
+const AXELAR_RELEASE_BASE_URL = 'https://static.axelar.network/releases/axelar-cgp-stellar';
+
+const SUPPORTED_CONTRACTS = new Set([
+    'axelar_gateway',
+    'axelar_operators',
+    'axelar_gas_service',
+    'interchain_token',
+    'token_manager',
+    'interchain_token_service',
+    'upgrader',
+]);
+
+function getWasmUrl(contractName, version) {
+    if (!SUPPORTED_CONTRACTS.has(contractName)) {
+        throw new Error(`Unsupported contract ${contractName} for versioned deployment`);
+    }
+
+    const pathName = contractName.replace(/_/g, '-');
+
+    return `${AXELAR_RELEASE_BASE_URL}/stellar-${pathName}/${version}/wasm/stellar_${contractName}.wasm`;
+}
+
+async function downloadWasmFile(contractName, version) {
+    const url = getWasmUrl(contractName, version);
+    const tempDir = path.join(process.cwd(), 'artifacts');
```

Comment:
`fs` has an ability to create tempfiles, take a look at that perhaps?
https://github.com/axelarnetwork/axelar-cgp-sui/blob/81888b829095522d3d728527398bf63a5dfc5d5b/src/node/node-utils.ts#L15